File: Python/level2/짝지어 제거하기.py
def solution(s):
    stack = []
    for string in s:
        if len(stack) == 0:
            stack.append(string)
        else:
            prevString = stack.pop()
            if prevString != string:
                stack += [prevString, string]
    return 1 if len(stack) == 0 else 0

File: Cplusplus/LEVEL 2/짝지어 제거하기.cpp
//22.01.28 짝지어 제거하기 by CJHofficial

#include <iostream>
#include <string>
#include <stack>
using namespace std;

int solution(string s)
{
    stack<char> st;
    for (auto c : s) {
        if (!st.empty()) {
            if (st.top() == c) st.pop();
            else st.push(c);
        }
        else st.push(c);
    }
    return (st.empty()) ? 1 : 0;
}